react/ProductSummaryImage.tsx

```diff
@@ -87,11 +95,12 @@ function getHoverImage({
   hoverImage,
   hoverImageLabel,
 }: GetHoverImageParams) {
-  const { criteria = 'label', label = hoverImageLabel, index } =
-    hoverImage ?? {}
+  const { criteria = 'label', index } = hoverImage ?? {}
+
+  const label = hoverImage?.label ?? hoverImageLabel
 
   if (criteria === 'label') {
-    return findImageByLabel(images, label)
+    return findImageByLabel(images, label, hoverImage?.labelMatchCriteria)
```

Comment:
Also, make sure to update your references to this variable, since they'll be available by just calling `labelMatchCriteria` instead of `hoverImage?.labelMatchCriteria`.

Reply:
@icazevedo Done, also updated the references